wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/Extend.kt

```diff
@@ -38,9 +39,14 @@ class Extend private constructor(
     }
   }
 
-  fun validate(linker: Linker) {
+  fun validate(linker: Linker, syntaxRules: SyntaxRules) {
     val linker = linker.withContext(this)
     linker.validateImport(location, type!!)
+
+    if (!syntaxRules.allowsMessageExtensions() &&
+        ProtoType.get(name) !in GOOGLE_PROTOBUF_OPTION_TYPES) {
```

Comment:
I was thinking, how about we try passing the proto type into the method? would that be such a bad thing...?

Reply:
Yeah what do you think about ? I think that's good

```kotlin
// proto2
fun allowsExtensions(type: ProtoType) = true
// proto3
fun allowsExtensions(type: ProtoType) {
  return type in GOOGLE_PROTOBUF_OPTION_TYPES
}
```

Reply:
[memo]
SyntaxRule is not just a config, e.g. a list of booleans for a particular syntax rule. It can have some logic to tell the API user whether or not a specific syntax is met.